sklearn/linear_model/coordinate_descent.py

```diff
@@ -1795,7 +1795,7 @@ def fit(self, X, y):
         X, y, X_offset, y_offset, X_scale = _preprocess_data(
             X, y, self.fit_intercept, self.normalize, copy=False)
 
-        if not self.warm_start or self.coef_ is None:
+        if not self.warm_start or not hasattr(self, "coef_"):
```

Comment:
Just to be safe, we might want to  handle the case where coef_ is None in case someone relied on this broken implementation...?

Reply:
Something like 
```suggestion
if not self.warm_start or not hasattr(self, "coef_") or self.coef_ is None:
```
could handle that situation, right?

Reply:
Maybe this condition
```suggestion
        if not self.warm_start or not hasattr(self, "coef_") or self.coef_ is None:
```
should be able to handle that?